### ISSUE TITLE
Rename metric to scoring for better sklearn compatibility

### ISSUE THREAD

**Issue by mfeurer** [automl/auto-sklearn#1457]
Scikit-learn's model selection modules such as [RandomizedSearchCV](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.RandomizedSearchCV.html#sklearn.model_selection.RandomizedSearchCV) and [HalvingRandomSearchCV](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.HalvingRandomSearchCV.html#sklearn.model_selection.HalvingRandomSearchCV) use the feature name `scoring` instead of `metric`. We should follow scikit-learn here and deprecate the argument `metric` and replace it by `scoring`.
